### PR TITLE
initial flask_api tests

### PR DESCRIPTION
**Pull request by Jason-Ku**
## What's New?
* Add some starter tests for the flask app
* Rebased
* Added is_view variable to Entity classes
* Check if an entity is a view before update/insert
* Added test for view check

## How Has This Been Tested?
`sh run_tests.sh`
image

## Checklist (check-all-before-merge)
_formatting help: `- [x]` means "checked' and `- [ ]` means "unchecked"_

- [ ] I documented my code according to the [Google Python Style Guide][1]

- [ ] I ran `./build_docs.sh` and the docs look fine

- [ ] I ran `./type_check.sh` and got no errors

- [ ] I ran `./format.sh` because it automatically cleans my code for me 😄 

- [ ] I ran `./lint.sh` to check for what "format" missed

- [x] I added my tests to the `/tests` directory

- [x] I ran `./run_tests.sh` and all the tests pass

[1]: https://google.github.io/styleguide/pyguide.html

### CODE REVIEW

File: flask_api.py [hunk 1]
```diff
@@ -45,9 +45,16 @@
 app = Flask(__name__)
 CORS(app)
 
-# TODO: Initialize these somewhere else
-db = NimbusMySQLAlchemy()
-nimbus = Nimbus(db)
+db = None
```
Comment (Jason-Ku) [inline flask_api.py:48]: went with the approach that @cameron-toy proposed, I agree that this singleton-esque way to handle it is pretty good

File: nimbus_nlp/question_classifier.py [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+from nimbus_nlp.save_and_load_model import save_model, load_latest_model, PROJECT_DIR
+from typing import Tuple
 
 # TODO: move the functionality in this module into class(es), so that it can be more easily used as a dependency
 
 
 class QuestionClassifier:
-    def __init__(self, file_path="question_set_clean.csv"):
+    def __init__(self, db, file_path="question_set_clean.csv"):
+        self.db = db
```
Comment (Jason-Ku) [inline nimbus_nlp/question_classifier.py:15]: If using the DI here, need to update the other places where QuestionClassifier is constructed

File: tests/test_flask_api.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+@pytest.fixture
+def client():
+    flask_api.app.config['TESTING'] = True
+
+    with flask_api.app.test_client() as client:
+        yield client
+
+
+def test_hello(client):
```
Comment (Jason-Ku) [inline tests/test_flask_api.py:25]: Not a particularly helpful test but I figured I'd add it so other tests can build off this one

File: flask_api.py [hunk 4]
```diff
@@ -118,7 +126,7 @@ def save_a_recording():
     # Let's also save the filename to the database for quick reference
     formatted_data["filename"] = filename
 
-    db = NimbusMySQLAlchemy(config_file=CONFIG_FILE_PATH)
+    init_nimbus()
```
Comment (Jason-Ku) [inline flask_api.py]: now we won't create a new db instance everytime an endpoint is called

File: nimbus.py [hunk 5]
```diff
@@ -19,7 +19,7 @@ def __init__(self, db: NimbusMySQLAlchemy):
         self.qa_dict = create_qa_mapping(generate_qa_pairs(qa_pairs, db))
         # Instantiate variable extractor and question classifier
         self.variable_extractor = VariableExtractor()
-        self.classifier = QuestionClassifier()
+        self.classifier = QuestionClassifier(db)
```
Comment (Jason-Ku) [inline nimbus.py:22]: reflects the change in question_classifier.py, a quick search shows that this is the only use of QuestionClassifier, so no need to worry about broken stuff

**Review by chidiewenike — APPROVED** [review]
Looks good!

File: tests/test_flask_api.py (hunk 3 above)
Comment (snekiam) [reply tests/test_flask_api.py:25]: This does test for a basic problem with flask starting, I think its reasonable.

**Review by snekiam — APPROVED** [review]
Looks great!

**Comment by Jason-Ku** [comment]
@cameron-toy does the init db/nimbus part differently, which I have no problem with, I may update this PR to follow his model

**Comment by chidiewenike** [comment]
@cameron-toy Does this conflict with your most recent PR? I do believe that this could be the solution to the memory leak.

**Review by snekiam — APPROVED** [review]

**Pull request merged**